Restore and update Payment Services dropin docs (#852)
commerce-docs/microsite-commerce-storefront: src/content/docs/dropins/payment-services/containers/apple-pay.mdx

  label: ApplePay
---

import Link from '@components/Link.astro';
import TableWrapper from '@components/TableWrapper.astro';

> The `ApplePay` container renders a button that macOS/iOS users can use to check out using <Link href="https://www.apple.com/apple-pay/" text="Apple Pay" />.

Reviewer: ```suggestion
The `ApplePay` container renders a checkout button that enables macOS and iOS users to pay using <Link href="https://www.apple.com/apple-pay/" text="Apple Pay" />.
```

Author: @bdenham to backtick-quote or not to backtick-quote 🤔

I see you removed the backticks from the `ApplePay` component name but kept them for `CreditCard` component in `credit-card.mdx`.

Reviewer: Oops. I'll add them back to ApplePay

<div style="background-color: var(--sl-color-blue-low); border-left: 4px solid var(--sl-color-blue); padding: 0.75rem 1rem; border-radius: 0.25rem; margin: 1rem 0;">
<strong>Version: 3.1.1</strong>
</div>

## Configuration

The `ApplePay` container provides the following configuration options:

<TableWrapper nowrap={[0, 1, 2]}>
| Option | Type | Req? | Description |
|--------|------|------|-------------|

> | `location` | string | Yes | Location where the Apple Pay is to be rendered. Must be either `CHECKOUT` or `PRODUCT_DETAIL`. |

Reviewer: ```suggestion
| `location` | string | Yes | Location where Apple Pay is rendered. Must be either `CHECKOUT` or `PRODUCT_DETAIL`. |
```

> | `getCartId` | function | Maybe | Required if `createCart` not provided. Should return a promise that resolves to the shopper's cart ID. |

Reviewer: ```suggestion
| `getCartId` | function | Maybe | Required if `createCart` is not provided. Returns a promise that resolves to the shopper's cart ID. |
```

> | `createCart` | object | Maybe | Required if `getCartId` not provided. Provides cart items when `getCartId` is not used. The object must define a single `getCartItems` property, which must be a function that returns a list of cart items to purchase. The list must contain at least one cart item. Cart items must be provided as objects with at minimum a `sku` (string) and a `quantity` (number). See the "Cart item object" section for more information. |

Reviewer: ```suggestion
| `createCart` | object | Maybe | Required if `getCartId` is not provided. Provides cart items when `getCartId` is not used. Must be an object with a `getCartItems` function that returns at least one cart item. Each item must include at least a `sku` (string) and `quantity` (number). See "Cart item object" for details. |
```

> | `onButtonClick` | function | No | Called when the shopper clicks the Apple Pay button. The callback receives a `showPaymentSheet` function as its only argument that must be called to begin the Apple Pay session and show the payment sheet. If no `onButtonClick` callback is provided, clicking the Apple Pay button will automatically trigger the payment sheet. Note that the 'showPaymentSheet' callback **must** be called synchronously. |

Reviewer: ```suggestion
| `onButtonClick` | function | No | Called when the shopper clicks the Apple Pay button. Receives a `showPaymentSheet` function, which must be called synchronously to start the Apple Pay session and display the payment sheet. If the `onButtonClick` callback is not provided, clicking the Apple Pay button will automatically trigger the payment sheet. |
```

> | `onSuccess` | function | No | Called when the payment flow finishes successfully. The function receives an `{ cartId: string }` object as an argument. If the function returns a promise, the promise will be awaited before marking the payment as "completed successfully" in the Apple Pay sheet. If the promise rejects, the payment will be marked as "failed" in the Apple Pay sheet and the error will be propagated to the `onError` callback, if provided. |

Reviewer: ```suggestion
| `onSuccess` | function | No | Called when the payment completes successfully. Receives `{ cartId: string }`. If the function returns a promise, it is awaited before marking the payment as successful in the Apple Pay sheet. If the promise rejects, the payment is marked as failed and the error is passed to `onError` (if provided). |
```

> | `onError` | function | No | Called when the payment flow was aborted due to an error. The function receives an object with two properties: `{name: string, message: string}`, containing the localized error name and message. Both properties are user-facing and can be translated using the `PaymentServices.ApplePay.errors` [language definitions](/dropins/payment-services/dictionary/). |

Reviewer: ```suggestion
| `onError` | function | No | Called when the payment flow fails or is aborted. Receives `{ name: string, message: string }`, containing localized, user-facing error details. These values can be translated using `PaymentServices.ApplePay.errors` [language definitions](/dropins/payment-services/dictionary/). |
```

> | `hidden` | boolean | No | Whether the button is hidden. Set this to `true` to hide the Apple Pay button (default: `false`). |

Reviewer: ```suggestion
| `hidden` | boolean | No | Whether the button is hidden. Set to `true` to hide the Apple Pay button. Default: `false`. |
```

> | `disabled` | boolean | No | Whether the button is disabled. Set this to `true` to disable the Apple Pay button (default: `false`). |

Reviewer: ```suggestion
| `disabled` | boolean | No | Whether the button is disabled. Set to `true` to disable the Apple Pay button. Default: `false`. |
```
